### ISSUE TITLE
issue on limit the range of bounding box regression

### ISSUE THREAD

**Issue by neocats** [Stick-To/FCOS-tensorflow#3]
if self.mode == 'train':
                total_loss = []
                # ground_truth [batch, num_box, ymin, ymax, xmin, xmax, classid]
                for i in range(self.batch_size):
                    gt_i = self.ground_truth[i, ...]
                    slice_index = tf.argmin(gt_i, axis=0)[0]
                    gt_i = tf.gather(gt_i, tf.range(0, slice_index, dtype=tf.int64))
                    gt_size = tf.sqrt(gt_i[..., 2] * gt_i[..., 3])
                    g3 = tf.boolean_mask(gt_i, gt_size <= 64.)

#############
I found this part maybe different form origin paper. I dont't get the point of "slice_index".
For your implement, if gt_i contain just one bbox , “slice_index = tf.argmin(gt_i, axis=0)[0]” will be equal zero, so “gt_i ” in next line will gather nothing(for tf.range(0, 0)).  
Or am I have some misunderstanding? THX

**Comment by Stick-To** [comment]
for input the ground_truth as a batch, I pad the all ground_truth to 60 (for voc) with [-1, -1, -1, -1, -1],
[batch_size, y, x, h, w, classid].

**Comment by Stick-To** [comment]
the ground_truth input is batchx60x5 now

**Comment by neocats** [comment]
Thank you for your prompt reply.
Another question,  I found the origin paper use the maximum value of the distance (l, t, r, b) to the bbox boundary at eatch point in the feature map to assing the positive and negative sample, but in your implement maybe you straightforward use the bbox‘s square to do that job. What's the benefit of doing this？ 
Look forward to your kind advice.

**Comment by Stick-To** [comment]
it is the minimum area, I implement the FCOS follwe the original paper

**Comment by neocats** [comment]
Maybe I have some misunderstanding, :D THX

**Comment by Stick-To** [comment]
I split postive and negtive using mask ----------variable loc in line 298

**Comment by neocats** [comment]
Thanks a lot! :D

**Comment by chenyipeng1** [comment]
I think what you think about minimum area is wrong. This is not equal to the max distance of (l,t,r,b)*. Though it may not perform big difference in AP, but I think you are wrong.

**Comment by Stick-To** [comment]
@chenyipeng1 ， computing the max distance of (l,t,r,b)* is for computing iou loss,  computing minimum area is wrong is in front， rows 287-302

**Comment by Stick-To** [comment]
@chenyipeng1 ， computing the max distance of (l,t,r,b)* is for computing iou loss, computing minimum area is in front rows 278-302
